zerver/actions/user_groups.py

```diff
@@ -278,6 +279,19 @@ def add_subgroups_to_user_group(
     GroupGroupMembership.objects.bulk_create(group_memberships)
 
     subgroup_ids = [subgroup.id for subgroup in subgroups]
+    now = timezone_now()
+    RealmAuditLog.objects.bulk_create(
+        RealmAuditLog(
+            realm=user_group.realm,
+            modified_user_group=user_group,
+            event_type=RealmAuditLog.USER_GROUP_DIRECT_SUBGROUP_MEMBERSHIP_ADDED,
+            event_time=now,
+            acting_user=acting_user,
+            extra_data=orjson.dumps({"subgroup_id": subgroup_id}).decode(),
```

Comment:
We can possible have `extra_data` store a list of subgroup IDs with `subgroup_ids` as key instead of different objects for each subgroup, but I am not sure what would be better. Would wait for feedback from Tim on this.

Reply:
Sidenote: perhaps it can be similar to what we did for the realm filter events, by having a complete set of subgroups there are at that point. But it might become very clumsy at some point.